File: TTSPrep_API/Models/TextBlock.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TTSPrep_API.Models;

public class TextBlock
{
    [Key]
    public string Id { get; set; }
    public string? Label { get; set; }
    [Range(1, int.MaxValue)]
    public int OrderNumber { get; set; }
    public string? OriginalText { get; set; }
    public string? ModifiedText { get; set; }
    [ForeignKey(nameof(Chapter))]
    public string ChapterId { get; set; }
}

File: ttsprep-client/src/app/models/textBlock.ts
import {Speaker} from "./speaker";
import {TextBlockLabel} from "./textBlockLabel";

export interface TextBlock {
  id: string
  orderNumber: number
  originalText?: string
  modifiedText?: string
  chapterId: string
  speakerId?: string
  speaker?: Speaker
  textBlockLabelId?: string
  textBlockLabel?: TextBlockLabel
}
